src/main/java/org/folio/dataimport/util/Try.java

```diff
@@ -0,0 +1,54 @@
+package org.folio.dataimport.util;
+
+import io.vertx.core.Future;
+
+import java.util.function.Supplier;
+
+/**
+ * Util class which contains methods with boiler-plate code for exceptions handling under async methods calls
+ */
+public class Try {
+
+  private Try() {
+  }
+
+  /**
+   * Return future with result returned by specified task.
+   * If the {@code task} throws an exception, the returned future will be failed with this exception
+   *
+   * @param task task
+   * @return future with result from {@code task} execution
+   */
+  public static <T> Future<T> itGet(Supplier<T> task) {
+    Future<T> future = Future.future();
+    try {
+      future.complete(task.get());
+    } catch (Exception e) {
+      future.fail(e);
+    }
+    return future;
+  }
+
+  /**
+   * Return succeed or failed future handled by specified job.
+   * If the {@code job} throws an exception, the returned future will be failed with this exception
+   *
```

Comment:
I sugest:
Executes a job. It creates new future and pass it to a job as a parameter, so you are able to control it. If a job is failed then the future is completed and propagated with an exception.